app/models/reservation.rb

```diff
@@ -178,6 +178,11 @@ def force_dirty!
   end
 
   def start_reservation!
+    # Auto-end functionality: automatically end previous reservations when feature is enabled
+    if SettingsHelper.feature_on?(:auto_end_reservations_on_next_start)
+      AutoEndPreviousReservation.new(product, user).end_previous_reservations!
+    end
```

Comment:
Can we move this to the controller or to an active job and call it from the controller?

I think this model method is too complex already to add this side effect.
Also, since this is a side effect of start a reservation I think any errors occurred within the new code shouldn't affect the original functionallity